include/rocksdb/filter_policy.h

```diff
@@ -59,19 +59,32 @@ class FilterBitsBuilder {
 
   // Generate the filter using the keys that are added
   // The return value of this function would be the filter bits,
-  // The ownership of actual data is set to buf
-  virtual Slice Finish(std::unique_ptr<const char[]>* buf) = 0;
-
-  // Similar to Finish(std::unique_ptr<const char[]>* buf), except that
-  // for a non-null status pointer argument, it will point to
-  // Status::Corruption() when there is any corruption during filter
-  // construction or Status::OK() otherwise.
+  // The ownership of actual data is set to buf.
+  // RocksDB does not call this function in production except via the default
+  // implementation of FinishV2. The default implementation of this function
+  // calls FinishV2, so one of Finish and FinishV2 must be overridden (to
+  // avoid call stack overflow).
+  virtual Slice Finish(std::unique_ptr<const char[]>* buf);
+
+  // Newer version of Finish() that supports
+  // * Returning a Status, in case an unexpected memory corruption is detected
+  // in filter construction.
+  // * Allocating the returned buffer with a custom allocator (such as for
+  // cache warming during table construction)
   //
-  // WARNING: do not use a filter resulted from a corrupted construction
-  virtual Slice Finish(std::unique_ptr<const char[]>* buf,
-                       Status* /* status */) {
-    return Finish(buf);
-  }
+  // If returning Status::OK() and output_filter->size() > 0, then
+  // output_filter->data() points to a buffer allocated by `allocator` (or
+  // default allocator with allocator==nullptr). output_filter->size() might
+  // be larger than the original requested allocation size if
+  // allocator->UsableSize() says the space is usable. (See
+  // optimize_filters_for_memory.)
+  //
+  // If not returning Status::OK(), then `output_filter` should be unmodified
+  // or reset to Slice(). (Caller is not responsible for any deallocation.)
```

Comment:
Is "(Caller is not responsible for any deallocation.)" referring to the case:  "if `output_filter` (I meant the actual slice pointed by `Slice* output_filter`) is reset to Slice() (i.e, ), then caller is not responsible for any deallocation of the original content of that slice"? 

If it is, I thought the caller would never need to consider such deallocation using Slice. For example:
```
std::string a_str("abc");
Slice a(a_str, a_str.size());
std::string b_str("efg");
Slice b(b_str, b_str.size());
a = b; // Doing this does not require any deallocation of a_str
```